<a href="https://colab.research.google.com/github/hasanocal7/DS-Libs/blob/main/Pandas2_Odev.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy as np
import pandas as pd

In [120]:
df_tips = pd.read_csv("tips.csv")
df_tips

total_bill   tip     sex smoker   day    time  size  price_per_person  \
0         16.99  1.01  Female     No   Sun  Dinner     2              8.49   
1         10.34  1.66    Male     No   Sun  Dinner     3              3.45   
2         21.01  3.50    Male     No   Sun  Dinner     3              7.00   
3         23.68  3.31    Male     No   Sun  Dinner     2             11.84   
4         24.59  3.61  Female     No   Sun  Dinner     4              6.15   
..          ...   ...     ...    ...   ...     ...   ...               ...   
239       29.03  5.92    Male     No   Sat  Dinner     3              9.68   
240       27.18  2.00  Female    Yes   Sat  Dinner     2             13.59   
241       22.67  2.00    Male    Yes   Sat  Dinner     2             11.34   
242       17.82  1.75    Male     No   Sat  Dinner     2              8.91   
243       18.78  3.00  Female     No  Thur  Dinner     2              9.39   

             Payer Name         CC Number Payment ID  
0    Christy Cunningham  3560325168603410    Sun2959  
1        Douglas Tucker  4478071379779230    Sun4608  
2        Travis Walters  6011812112971322    Sun4458  
3      Nathaniel Harris  4676137647685994    Sun5260  
4          Tonya Carter  4832732618637221    Sun2251  
..                  ...               ...        ...  
239       Michael Avila  5296068606052842    Sat2657  
240      Monica Sanders  3506806155565404    Sat1766  
241          Keith Wong  6011891618747196    Sat3880  
242        Dennis Dixon     4375220550950      Sat17  
243     Michelle Hardin  3511451626698139    Thur672  

[244 rows x 11 columns]

In [121]:
#1
#a
def last_3(x):
  return x[-3:]

In [122]:
df_tips["Payment ID"].apply(last_3)

0      959
1      608
2      458
3      260
4      251
      ... 
239    657
240    766
241    880
242    t17
243    672
Name: Payment ID, Length: 244, dtype: object

In [123]:
#b
df_tips["Payment ID"].apply(lambda x : x[-3:])

0      959
1      608
2      458
3      260
4      251
      ... 
239    657
240    766
241    880
242    t17
243    672
Name: Payment ID, Length: 244, dtype: object

In [124]:
#c
df_tips["Payment ID"].astype(str).str[-3:]

0      959
1      608
2      458
3      260
4      251
      ... 
239    657
240    766
241    880
242    t17
243    672
Name: Payment ID, Length: 244, dtype: object

In [141]:
#2
df_tips["Just Name"] = df_tips["Payer Name"].astype(str).str.split(" ").str[0] 

In [126]:
df_tips.head()

total_bill   tip     sex smoker  day    time  size  price_per_person  \
0       16.99  1.01  Female     No  Sun  Dinner     2              8.49   
1       10.34  1.66    Male     No  Sun  Dinner     3              3.45   
2       21.01  3.50    Male     No  Sun  Dinner     3              7.00   
3       23.68  3.31    Male     No  Sun  Dinner     2             11.84   
4       24.59  3.61  Female     No  Sun  Dinner     4              6.15   

           Payer Name         CC Number Payment ID  Just Name  
0  Christy Cunningham  3560325168603410    Sun2959    Christy  
1      Douglas Tucker  4478071379779230    Sun4608    Douglas  
2      Travis Walters  6011812112971322    Sun4458     Travis  
3    Nathaniel Harris  4676137647685994    Sun5260  Nathaniel  
4        Tonya Carter  4832732618637221    Sun2251      Tonya

In [142]:
#3
df_tips["Just Name"].value_counts()

Jason          9
Michael        7
James          6
David          6
Christopher    5
              ..
Shawna         1
Louis          1
Chris          1
Mark           1
Michelle       1
Name: Just Name, Length: 153, dtype: int64

In [143]:
#4
df_tips["Payer Name"].value_counts() # Görüldüğü üzere herkes benzersiz 3. soruda yapılan şey isim benzerliklerinin sayısal kıyaslanması idi

Christy Cunningham    1
Todd Patterson        1
Madison Wilson        1
Ryan Gonzales         1
Laura Robles          1
                     ..
Shawna Cook           1
Derek Thomas          1
Donald Williams       1
Roger Taylor          1
Michelle Hardin       1
Name: Payer Name, Length: 244, dtype: int64

In [144]:
df_tips = df_tips.drop("Just Name" , axis = 1)

In [145]:
df_tips.head()

total_bill   tip     sex smoker  day    time  size  price_per_person  \
0       16.99  1.01  Female     No  Sun  Dinner     2              8.49   
1       10.34  1.66    Male     No  Sun  Dinner     3              3.45   
2       21.01  3.50    Male     No  Sun  Dinner     3              7.00   
3       23.68  3.31    Male     No  Sun  Dinner     2             11.84   
4       24.59  3.61  Female     No  Sun  Dinner     4              6.15   

           Payer Name         CC Number Payment ID  
0  Christy Cunningham  3560325168603410    Sun2959  
1      Douglas Tucker  4478071379779230    Sun4608  
2      Travis Walters  6011812112971322    Sun4458  
3    Nathaniel Harris  4676137647685994    Sun5260  
4        Tonya Carter  4832732618637221    Sun2251

In [150]:
#5
df_male = df_tips[df_tips.sex == "Male"]
df_female = df_tips[df_tips.sex == "Female"]

In [155]:
df_male.shape[0]

157

In [156]:
df_female.shape[0]

87

In [161]:
#6
df_male.tip.sum() - df_female.tip.sum() # Erkekler daha çok bahşiş ödemiş

238.56000000000003

In [162]:
#7
df_male.tip.mean() - df_female.tip.mean() # Erkekler her türlü fazla ödemiş :D

0.25616955853283585

In [178]:
#8
df_male.smoker.value_counts()

No     97
Yes    60
Name: smoker, dtype: int64

In [189]:
a = df_male.smoker.value_counts().values / df_male.shape[0] * 100
print(f"Sigara İçen Erkeklerin Yüzdesi : {a[1]}\nSigara İçmeyen Erkeklerin Yüzdesi {a[0]}")

Sigara İçen Erkeklerin Yüzdesi : 38.21656050955414
Sigara İçmeyen Erkeklerin Yüzdesi 61.78343949044586


In [180]:
df_female.smoker.value_counts()

No     54
Yes    33
Name: smoker, dtype: int64

In [188]:
a = df_female.smoker.value_counts().values / df_female.shape[0] * 100
print(f"Sigara İçen Kadınların Yüzdesi : {a[1]}\nSigara İçmeyen Kadınların Yüzdesi {a[0]}")

Sigara İçen Kadınların Yüzdesi : 37.93103448275862
Sigara İçmeyen Kadınların Yüzdesi 62.06896551724138


In [182]:
# Neredeyse yüzdelik olarak birbirine yakın olsalar da erkekler daha çok içiyor...

In [183]:
#9
df_male.head()

total_bill   tip   sex smoker  day    time  size  price_per_person  \
1       10.34  1.66  Male     No  Sun  Dinner     3              3.45   
2       21.01  3.50  Male     No  Sun  Dinner     3              7.00   
3       23.68  3.31  Male     No  Sun  Dinner     2             11.84   
5       25.29  4.71  Male     No  Sun  Dinner     4              6.32   
6        8.77  2.00  Male     No  Sun  Dinner     2              4.38   

           Payer Name         CC Number Payment ID  
1      Douglas Tucker  4478071379779230    Sun4608  
2      Travis Walters  6011812112971322    Sun4458  
3    Nathaniel Harris  4676137647685994    Sun5260  
5          Erik Smith   213140353657882    Sun9679  
6  Kristopher Johnson  2223727524230344    Sun5985

In [184]:
df_male.time.value_counts()

Dinner    124
Lunch      33
Name: time, dtype: int64

In [200]:
a = df_male.time.value_counts().values / df_male.shape[0] * 100
print(f"Akşam Yemeği Yiyen Erkeklerin Yüzdesi : {a[0]}\nÖğlen Yemeği Yiyen Erkeklerin Yüzdesi : {a[1]}")

Akşam Yemeği Yiyen Erkeklerin Yüzdesi : 78.98089171974523
Öğlen Yemeği Yiyen Erkeklerin Yüzdesi : 21.019108280254777


In [196]:
df_female.time.value_counts()

Dinner    52
Lunch     35
Name: time, dtype: int64

In [199]:
a = df_female.time.value_counts().values / df_female.shape[0] * 100
print(f"Akşam Yemeği Yiyen Kadınların Yüzdesi : {a[0]}\nÖğlen Yemeği Yiyen Kadınların Yüzdesi : {a[1]}")

Akşam Yemeği Yiyen Kadınların Yüzdesi : 59.77011494252874
Öğlen Yemeği Yiyen Kadınların Yüzdesi : 40.229885057471265


In [204]:
# Her iki cinsiyette restoranta daha sık Akşam Yemeği vaktinde gitmeyi tercih ediyor

In [205]:
#10
df_tips.drop("CC Number" , axis = 1 , inplace = True)

In [206]:
df_tips.head()

total_bill   tip     sex smoker  day    time  size  price_per_person  \
0       16.99  1.01  Female     No  Sun  Dinner     2              8.49   
1       10.34  1.66    Male     No  Sun  Dinner     3              3.45   
2       21.01  3.50    Male     No  Sun  Dinner     3              7.00   
3       23.68  3.31    Male     No  Sun  Dinner     2             11.84   
4       24.59  3.61  Female     No  Sun  Dinner     4              6.15   

           Payer Name Payment ID  
0  Christy Cunningham    Sun2959  
1      Douglas Tucker    Sun4608  
2      Travis Walters    Sun4458  
3    Nathaniel Harris    Sun5260  
4        Tonya Carter    Sun2251

In [214]:
df_tips.corr()

total_bill       tip      size  price_per_person
total_bill          1.000000  0.675734  0.598315          0.647554
tip                 0.675734  1.000000  0.489299          0.347405
size                0.598315  0.489299  1.000000         -0.175359
price_per_person    0.647554  0.347405 -0.175359          1.000000



---

*   ***total_bill***, price_per_person ile güçlü bir pozitif korelasyona sahiptir (0.647554), yani toplam hesap arttıkça, kişi başına düşen fiyat da artar.

---


*   ***total_bill***, tip ile orta derecede güçlü pozitif bir korelasyona sahiptir (0.675734), yani toplam hesap arttıkça, bahşiş miktarı da artar.



---


*   ***size***, total_bill ile orta derecede güçlü pozitif bir korelasyona sahiptir (0.598315), yani daha büyük gruplar genellikle daha yüksek toplam hesaplara sahiptir.

---


*   ***size***, price_per_person ile zayıf negatif bir korelasyona sahiptir (-0.175359), yani grup büyüdükçe, kişi başına düşen fiyat azalmaya eğilimlidir.


---


*   ***tip***, price_per_person ile zayıf pozitif bir korelasyona sahiptir (0.347405), yani bahşiş miktarı arttıkça, kişi başına düşen fiyat da artar.



---




